### ISSUE TITLE
Implement `skorch`-based training pipeline

### ISSUE THREAD

**Issue by nclgbd** [nclgbd/ResearchToolKit#10]
# Overview

Use [`skorch`'s library](https://skorch.readthedocs.io/en/stable/index.html) to change training pipeline. This way, we can rely on `scikit-learn` for metric logging rather than `ignite`. `ignite` implementation is still important in #1, so this should be implemented as it's own module

## Subtasks

- [ ] Create metrics
    - This includes **AUC, Accuracy, Precision, Recall, and F1-score**. Loss may or may not be included, we'll see
- [ ] Create callbacks
    - A custom callback will need to be created for using **MLflow** and maybe AzureML we'll see, need to look further into the documentation
- [ ] Create `skorch` based training script.
    - The `ignite` based training should likewise be moved to another script for simplicity of code

## Additional resources
- https://skorch.readthedocs.io/en/stable/user/tutorials.html
- https://nbviewer.org/github/skorch-dev/skorch/blob/master/notebooks/Basic_Usage.ipynb
- https://github.com/skorch-dev/skorch/blob/master/notebooks/Transfer_Learning.ipynb
- https://nbviewer.org/github/skorch-dev/skorch/blob/master/notebooks/Hugging_Face_VisionTransformer.ipynb
- https://nbviewer.org/github/skorch-dev/skorch/blob/master/notebooks/Advanced_Usage.ipynb
- https://skorch.readthedocs.io/en/stable/skorch.html

**Comment by nclgbd** [comment]
Was a fun consideration, but not likely to implement. Too specific and too much additional implementation to warrant the pay off. But using `[torchmetrics](https://lightning.ai/docs/torchmetrics/stable//index.html)` seems to be a different more promising way to collect metrics, we'll see.
